src/ewatercycle_wflowjl/model.py

```diff
@@ -0,0 +1,307 @@
+    """Functionality for the Wflow.jl model."""
+
+    forcing: WflowJlForcing | None = None
+    parameter_set: ParameterSet
+
+    _config: dict = PrivateAttr()
+
+    @model_validator(mode="after")
+    def _initialize_config(self):
+        """Load config from parameter set and update with forcing info."""
+        cfg = toml.load(self.parameter_set.directory / self.parameter_set.config)
+
+        self._config = cfg
+
+    def _make_cfg_file(self, **kwargs) -> Path:
+        """Create a new wflow config file and return its path."""
+        if "start_time" in kwargs:
+            self._config["starttime"] = _iso_to_wflow(kwargs["start_time"])
+        if "end_time" in kwargs:
+            self._config["endtime"] = _iso_to_wflow(kwargs["end_time"])
+
+        # Get time specs from forcing
+        if self.forcing is not None:
+            if self.forcing.directory is not None:
+                forcing_path = self.forcing.directory / self.forcing.netcdfinput
+            else:
+                forcing_path = Path(self.forcing.netcdfinput)
+
+            ds = xr.open_dataset(forcing_path)
+            calendar_type = ds["time"].attrs.get("calendar_type")
+            if calendar_type is not None:
+                self._config["calendar"] = calendar_type
+            self._config["timestepsecs"] = int(
+                np.timedelta64(ds["time"].to_numpy()[1] - ds["time"].to_numpy()[0])
+                / np.timedelta64(1, "s")
+            )
+            if "start_time" not in kwargs:
+                self._config["starttime"] = _iso_to_wflow(self.forcing.start_time)
+            if "end_time" not in kwargs:
+                self._config["endtime"] = _iso_to_wflow(self.forcing.end_time)
+
+        config_file = self._cfg_dir / "wflow_ewatercycle.toml"
+
+        # Input paths
+        self._config["state"]["path_input"] = str(
+            Path(self.parameter_set.directory) / self._config["state"]["path_input"]
+        )
+        self._config["input"]["path_static"] = str(
+            Path(self.parameter_set.directory) / self._config["input"]["path_static"]
+        )
+        if self.forcing is None:
+            self._config["input"]["path_forcing"] = str(
+                Path(self.parameter_set.directory)
+                / self._config["input"]["path_forcing"]
+            )
+        else:
+            self._config["input"]["path_forcing"] = str(forcing_path)
+
+        # Output paths
+        self._config["state"]["path_output"] = str(
+            self._cfg_dir / self._config["state"]["path_output"]
+        )
+        self._config["output"]["path"] = str(
+            self._cfg_dir / self._config["output"]["path"]
+        )
+        self._config["csv"]["path"] = str(self._cfg_dir / self._config["csv"]["path"])
+        self._config["netcdf"]["path"] = str(
+            self._cfg_dir / self._config["netcdf"]["path"]
+        )
+
+        # Enable the following variables for the BMI:
+        self._config["API"]["components"] = [
```

Comment:
The config file I have was missing API section. I added created API section if it does not exist.

Reply:
Thanks! I should have caught that,